src/Client.cpp

```diff
@@ -1,65 +1,147 @@
-#include <stdio.h>
-#include <stdlib.h>
+#include <iostream>
+#include <inc/Client.hpp>
+#include <cstdio>
 #include <unistd.h>
-#include <string.h>
-#include <sys/types.h>
-#include <sys/socket.h>
-#include <netinet/in.h>
-#include <netdb.h>
+#include <poll.h>
 
-#define PORT 4000
+Client::Client(char _username[], char _serveraddr[], int _port)
+{
+  strcpy(m_username, _username);
+  m_socket = SocketClient(_serveraddr, _port);
+  inboxHasItem = false;
+  wait_server_response();
+};
 
-int main(int argc, char *argv[])
+void Client::client_controller()
 {
-  int sockfd, n;
-  struct sockaddr_in serv_addr;
-  struct hostent *server;
+  struct pollfd pfds[2];
 
-  char buffer[256];
-  if (argc < 2)
-  {
-    fprintf(stderr, "usage %s hostname\n", argv[0]);
-    exit(0);
-  }
+  char buff[256];
+  Message *msg = new Message();
 
-  server = gethostbyname(argv[1]);
-  if (server == NULL)
+  pfds[0].fd = STDIN_FILENO;
+  pfds[0].events = POLLIN;
+
+  pfds[1].fd = get_socket_num();
+  pfds[1].events = POLLIN;
+
+  uint16_t seq = 0;
+  while (1)
   {
-    fprintf(stderr, "ERROR, no such host\n");
-    exit(0);
+    if (poll(pfds, 2, 100 != -1))
+    {
+      if (pfds[0].revents & POLLIN) // message from stdin
+      {
+        cin >> buff;
+        fflush(stdin);
+        client_sender(buff);
+      }
+      if (pfds[1].revents & POLLIN) // received message from socket
+      {
+        client_receiver();
+      }
+      if (pfds[1].revents & (POLLERR | POLLHUP))
+      {
+        // socket was closed
+        cout << "oh no" << endl
```

Comment:
we probably need here a better message, it means we lost the socket connecting to the server